src/Evolution/Systems/Cce/InitializeCce.hpp

```diff
@@ -72,6 +92,153 @@ struct InitializeJ : public PUP::able {
       size_t number_of_radial_points) const noexcept = 0;
 };
 
+/*
+ * \brief Initialize \f$J\f$ on the first hypersurface by constraining
+ * \f$\psi_0 = 0\f$.
+ *
+ * \details This algorithm first radially evolves the \f$\psi_0 = 0\f$
+ * condition, which can be converted to a second-order radial ODE for J. Then,
+ *  the initial data generator performs an iterative solve for the angular
+ * coordinates necessary to ensure asymptotic flatness. The step size for the
+ * radial ODE integration is set by `RadialStep`; the parameters for the
+ * iterative procedure are determined by options
+ * `InitializeJZeroNonSmooth::AngularCoordinateTolerance` and
+ * `InitializeJZeroNonSmooth::MaxIterations`. The resulting `J` will necessarily
+ * have vanishing first radial derivative, and so will typically not be smooth
+ * (only continuous) with the provided Cauchy data at the worldtube boundary.
+ */
+struct InitializeJNoIncomingRadiation : public InitializeJ {
```

Comment:
Also this class looks like it's in the wrong commit. I'll stop reviewing for now and let you sort this out. Perhaps it would be helpful if you moved the more sophisticated algorithms for the `ZeroNonSmooth` and `NoIncomingRadiation` initial data into their own respective PRs to give them the attention they deserve